lib/elm/compiler.rb

```diff
@@ -5,34 +5,33 @@
 module Elm
   class Compiler
     class << self
-      def compile(elm_files, output_path: nil, elm_make_path: "elm-make", debug: false)
-        fail ExecutableNotFound unless elm_executable_exists?(elm_make_path)
+      def compile(elm_files, output_path: nil, elm_path: nil, debug: false)
+        elm_path ||= `which elm`.strip
+        fail ExecutableNotFound unless elm_executable_exists?(elm_path)
 
         if output_path
-          elm_make(elm_make_path, elm_files, output_path, debug)
+          elm_make(elm_path, elm_files, output_path, debug)
         else
-          compile_to_string(elm_make_path, elm_files, debug)
+          compile_to_string(elm_path, elm_files, debug)
         end
       end
 
       private
 
-      def elm_executable_exists?(elm_make_path)
-        Open3.popen2(elm_make_path){}.nil?
-      rescue Errno::ENOENT, Errno::EACCES
-        false
+      def elm_executable_exists?(elm_path)
+        File.exist?(elm_path)
```

Comment:
This doesn't check that it's a regular file or that it's executable by the current user.

Reply:
What would be a better way to check while throwing away any output?

Reply:
I couldn't figure out a nice way to toss the output with `Open3.popen2`

Reply:
There are a lot of `popen` variants. One should fit your needs.

Reply:
Ah you're right, `capture2` was perfect.